weldx/geometry.py

```diff
@@ -2439,8 +2452,107 @@ def from_file(file_name: Union[str, Path]) -> SpatialData:
+        tri_base = []
+        for i in range(num_profile_points):
+            idx_0 = i
+            idx_1 = (i + 1) % num_profile_points
+            idx_2 = idx_0 + num_profile_points
+            idx_3 = idx_1 + num_profile_points
+
+            if cw_ordering:
+                tri_base += [[idx_0, idx_2, idx_1], [idx_1, idx_2, idx_3]]
+            else:
+                tri_base += [[idx_0, idx_1, idx_2], [idx_1, idx_3, idx_2]]
+        tri_base = np.array(tri_base, dtype=int)
+
+        triangles = np.array(
+            [
+                tri_base + i * num_profile_points + offset
+                for i in range(num_profiles - 1)
+            ],
+            dtype=int,
+        )
+
+        return triangles.reshape((tri_base.shape[0] * (num_profiles - 1), 3)).tolist()
+
+    @staticmethod
+    def _shape_front_back_triangles(
+        num_profiles: int, num_profile_points: int, offset: int, cw_ordering: bool
+    ) -> List[List[int]]:
+        """Create the front and back surface triangles for ``_shape_triangles``."""
+        tri_cw = []
+        tri_ccw = []
+        i_0 = 0
+        i_1 = 0
+
+        while i_0 + i_1 < num_profile_points - 2:
+            p_0 = i_0 + offset
+            if i_1 == i_0:
+                p_1 = p_0 + 1
+                p_2 = num_profile_points + offset - i_1 - 1
+                i_0 += 1
+            else:
+                p_1 = num_profile_points + offset - i_1 - 2
+                p_2 = p_1 + 1
+                i_1 += 1
+            tri_cw += [[p_0, p_1, p_2]]
+            tri_ccw += [[p_0, p_2, p_1]]
+
+        if cw_ordering:
+            front = tri_cw
+            back = tri_ccw
+        else:
+            front = tri_ccw
+            back = tri_cw
+        return [
+            *front,
+            *(np.array(back, int) + (num_profiles - 1) * num_profile_points).tolist(),
+        ]
+
+    @classmethod
+    def _shape_triangles(
+        cls, shape_raster_data: np.ndarray, offset: int
+    ) -> List[List[int]]:
+        """Get the triangles of a shape from its raster data.
+
+        The triangle data are just indices referring to a list of points.
+
+        Parameters
+        ----------
+        shape_raster_data :
+            Raster data of the shape
+        offset :
+            An offset that will be added to all indices.
+
+        Returns
+        -------
+        List[List[int]] :
+            The list of triangles
+
+        """
+        n_prf = shape_raster_data.shape[0]
+        n_prf_pts = shape_raster_data.shape[1]
+        cw_ord = has_cw_ordering(shape_raster_data[0])
+
+        return [
+            *cls._shape_profile_triangles(n_prf, n_prf_pts, offset, cw_ord),
+            *cls._shape_front_back_triangles(n_prf, n_prf_pts, offset, cw_ord),
+        ]
+
+    @classmethod
     def from_geometry_raster(
-        geometry_raster: np.ndarray, closed_mesh: bool = True
+        cls, geometry_raster: np.ndarray, closed_mesh: bool = True
```

Comment:
the `closed_mesh` argument is not used in the code

Reply:
Right. The question is: Do we want to keep this option (easy to add to the new triangulation) or do we want to remove it?

Reply:
I thought the triangles would get set, it looks like that in the code? (line 2576)
Either way, would make sense to have the option

Reply:
Not sure if we are on the same boat here: The purpose of ´closed_mesh´ was to generate the additional side surfaces that were not covered by the initial algorithm. Now they are always generated but since it happens in its own method, we can just add a branch there. We just need to pass the option down to the method. I think it would be nice for the matplotlib plots since the additional triangles sometimes clutter the picture.

Reply:
ok I see, agree that it can be hand in some cases to not have those triangulations